snippets/PowerShell.json

```diff
@@ -952,5 +952,72 @@
             "#endregion"
         ],
         "description": "Region Block for organizing and folding of your code"
+    },
+    "IfShouldProcess": {
+        "prefix": "IfShouldProcess",
+        "body": [
+            "if (\\$PSCmdlet.ShouldProcess(\"${1:Target}\", \"${2:Operation}\")) {",
+                "\t$0",
+            "}"
+        ],
+        "description": "Adds ShouldProcess block"
+    },
+    "CalculatedProperty": {
+        "prefix": "Calculated-Property",
+        "body": [
+            "@{name='${1:PropertyName}';expression={\\$_.${2:PropertyValue}}}$0"
+        ],
+        "description": "Creates a PSCustomObject"
```

Comment:
I thought there was already a `PSCustomObject` snippet in this file. Is there not?

Also, isn't this just a `hashtable`, not a true `PSCustomObject`

Reply:
Woops. Thats my lazy copy paste it's a calculated property for use with select-object.

I'll update when back on the laptop.

Reply:
Updated the description for the calculated property @TylerLeonhardt

Reply:
Sorry can you give me an example of this? I'm having a hard time seeing the scenario.

Reply:
Linked from the `Select-Object` docs:

https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/select-object?view=powershell-6#examples

```markdown
-Property

Specifies the properties to select. These properties are added as NoteProperty members to the output objects. Wildcards are permitted.

The value of the Property parameter can be a new calculated property. To create a calculated, property, use a hash table. Valid keys are:

    Name (or Label) <string>
    Expression <string> or <script block>
```

```powershell
Get-Process | Select-Object -Property ProcessName,@{Name="Start Day"; Expression = {$_.StartTime.DayOfWeek}}

ProcessName  StartDay
----         --------
alg          Wednesday
ati2evxx     Wednesday
ati2evxx     Thursday
...
```

Reply:
I'd say it's used quite a bit when working with Active Directory.

```powershell
Get-ADUser -Identity brettmillerb -properties manager | Select-Object -Property @(
    @{name = 'Manager'; expression = { Get-ADUser -identity $_.Manager | Select-Object -ExpandProperty samaccountname }}
)
```

Reply:
Ahhh I see now! Thanks for the explanation.